[`eradicate`] ignore `# language=` in commented-out-code rule (ERA001)

fabiob opened the pull request:
Fixes one common case cited on #6019

<!--
Thank you for contributing to Ruff! To help us out with reviewing, please consider the following:

- Does this pull request include a summary of the change? (See below.)
- Does this pull request include a descriptive title?
- Does this pull request include references to any relevant issues?
-->

## Summary

The `commented-out-code` rule (ERA001) from `eradicate` is currently flagging a very common idiom that marks Python strings as another language, to help with syntax highlighting:

![image](https://github.com/user-attachments/assets/d523e83d-95cb-4668-a793-45f01d162234)

This PR adds this idiom to the list of allowed exceptions to the rule.

## Test Plan

I've added some additional test cases.

charliermarsh commented:
Do you mind pushing the test cases? I only see the code changes.

For reference: https://www.jetbrains.com/help/pycharm/using-language-injections.html

fabiob commented:
Sorry @charliermarsh , forgot to force-push after I amended the commit with the test cases.

charliermarsh commented:
No prob, thanks for the PR.

fabiob commented:
And looks like I need to fix the formatting... Working on it.

fabiob commented:
> For reference: https://www.jetbrains.com/help/pycharm/using-language-injections.html

I've added some more test cases and fixed the regexp with the additional cases described here.

charliermarsh commented:
Thanks and welcome to the project.

charliermarsh commented on crates/ruff_linter/src/rules/eradicate/detection.rs:
```diff
@@ -16,7 +16,7 @@ static CODE_INDICATORS: LazyLock<AhoCorasick> = LazyLock::new(|| {
 static ALLOWLIST_REGEX: LazyLock<Regex> = LazyLock::new(|| {
     Regex::new(
-        r"^(?i)(?:pylint|pyright|noqa|nosec|region|endregion|type:\s*ignore|fmt:\s*(on|off)|isort:\s*(on|off|skip|skip_file|split|dont-add-imports(:\s*\[.*?])?)|mypy:|SPDX-License-Identifier:|(?:en)?coding[:=][ \t]*([-_.a-zA-Z0-9]+))",
+        r"^(?i)(?:pylint|pyright|noqa|nosec|region|endregion|type:\s*ignore|fmt:\s*(on|off)|isort:\s*(on|off|skip|skip_file|split|dont-add-imports(:\s*\[.*?])?)|mypy:|SPDX-License-Identifier:|language=[a-zA-Z](?: ?[-_.a-zA-Z0-9]+)+(?:\s+prefix=\S+)?(?:\s+suffix=\S+)?|(?:en)?coding[:=][ \t]*([-_.a-zA-Z0-9]+))",
```
Should we just use `language=`? All language identifiers will start with that, and false positives seem somewhat rare / not a huge deal here.

fabiob replied:
@InSyncWithFoo maybe I was extra cautious about false positives.

> Technically, language IDs [can be anything](https://github.com/JetBrains/intellij-community/blob/986fff1b53ed8005bffae54a652d85c8bf901065/platform/core-api/src/com/intellij/lang/Language.java#L62),

Good catch! Maybe we can just require a non-space character after the equals sign. This should take care of most false positives anyway, as it's uncommon for Python users to omit spaces around assignment operators.

The only thing that worries me is about detecting a commented-out named parameter on a multi-line method invocation or declaration. But maybe I'm just being overzealous.

```python
def check_spelling(
  text,
  # language=DEFAULT_LANGUAGE
) ...
```

fabiob replied:
So @InSyncWithFoo , will you write the PR with the new regexp and test cases, or should I?
